api-server/test/conftest.py

```diff
@@ -29,6 +36,38 @@ async def api_client():
         yield client
 
 
+@pytest.fixture
+def mock_gh_client_factory():
+    mock_gh_client = Mock(spec=GithubClient)
+
+    def factory(
+        user_to_repos: Callable[..., list[RepositoryDict]],
+        repo_to_files: Callable[..., list[FileDict]],
+    ):
+        mock_gh_client.return_value.get_non_forked_repos = AsyncMock(
+            side_effect=user_to_repos
+        )
+
+        mock_gh_client.return_value.get_files_for_repo = AsyncMock(
+            side_effect=repo_to_files
+        )
+        return mock_gh_client
+
+    with patch("routes.session.GithubClient", new=mock_gh_client):
+        yield factory
```

Comment:
damn, what is this doing?

Reply:
This patches over the `GithubClient`  in [session.py](https://github.com/rohanshiva/gitgame/pull/46/files#diff-fde18b114663d01ae15b472096acb7d5dc4a13cf7e34d3c144deb2e8cfabcb9bR63). If we don't patch over, then the tests will actually hit the Github API.

Reply:
In addition to patching it over, in some tests, we may want to mock the response of the `GithubClient`. In order to do that, I made this a factory, a function that takes in the input parameters and returns an object, for the mock `GithubClient`. Check this [test](https://github.com/rohanshiva/gitgame/pull/46/files#diff-2b47e1807b689f4e10dce77b5b8f3dfcd237faabcf101f6a311734d37425a007R117) out to see how to use this fixture.